upup/pkg/fi/cloudup/networking.go

```diff
@@ -57,3 +58,20 @@ func usesCNI(c *api.Cluster) bool {
 	glog.Warningf("Unknown networking mode configured")
 	return true
 }
+
+// TODO: we really need to sort this out:
+// https://github.com/kubernetes/kops/issues/724
+// https://github.com/kubernetes/kops/issues/626
+// https://github.com/kubernetes/kubernetes/issues/30338
+const defaultCNIAsset = "https://storage.googleapis.com/kubernetes-release/network-plugins/cni-07a8a28637e97b22eb8dfe710eeae1344f69d16e.tar.gz"
+const defaultCNIAssetHashString = "19d49f7b2b99cd2493d5ae0ace896c64e289ccbb"
+const CNI_VERSION_URL = "CNI_VERSION_URL"
+
+func findCNIAssets(c *api.Cluster) (string, string) {
+
+	if cniVersionURL := os.Getenv(CNI_VERSION_URL); cniVersionURL != "" {
+		return cniVersionURL, ""
```

Comment:
Would be great to `glog.Infof("Using CNI version %q, as set in %s", cniVersionURL, CNI_VERSION_URL)`